### PR TITLE
Refactor learn ajax

### PR DESCRIPTION
**Pull request by cstillitano**
- move deck preview into users dir
- refactor learn ajax and setup sample fill in the blank prompt

### CODE REVIEW

File: app/controllers/learn_controller.rb [hunk 1]
```diff
@@ -4,23 +4,28 @@ class LearnController < ApplicationController
   def index
     @user = User.find(params[:id])
     @deck = Deck.find(params[:deck_id])
+    @prompt = Learn.series(@user, @deck)[:cards][2]
     @series = Learn.series(@user, @deck)
+
+    respond_to do |format|
+      format.html
+    end
```
Comment (findandrew) [inline app/controllers/learn_controller.rb]: have you confirmed that this respond_to is necessary within index?

Comment (cstillitano) [inline app/controllers/learn_controller.rb]: confirmed it is not necessary - will remove

File: app/views/users/_dashboard_deck.html.erb [hunk 2]
```diff
@@ -1,11 +1,11 @@
 <div class="block">
   <div class="block-image">
-    <%= link_to "/learn/#{@user.id}/#{deck.id}" do %>
+    <%= link_to "/users/#{@user.id}/#{deck.id}" do %>
       <%=image_tag image_path("temp/dashboard-img-#{rand(1..4)}.jpg") %>
     <% end %>
   </div>
   <div class="block-details">
-    <%= link_to "/learn/#{@user.id}/#{deck.id}", class: 'title-link' do %>
+    <%= link_to "/users/#{@user.id}/#{deck.id}", class: 'title-link' do %>
```
Comment (findandrew) [inline app/views/users/_dashboard_deck.html.erb:8]: there should be a url_helper for this no? `rake routes | grep user`

File: app/views/learn/_post_user_response.html.erb [hunk 3]
```diff
@@ -2,11 +2,10 @@
   <div class="row">
     <div class="col-sm-12 col-md-8 col-md-offset-2 learn-panel reponse">
       <div class="content">
-        <h1>Response</h1>
         <p>Correct! -or- Incorrect!</p>
       </div>
       <div class="controls">
-        <%= link_to 'Next Prompt', "/learn/#{@user.id}/#{@deck.id}/prompt", remote: true, class: 'btn btn-brand btn-outline' %>
+        <%= link_to 'Next Prompt', "/learn/#{@user.id}/#{@deck.id}/get-card-prompt", remote: true, class: 'btn btn-brand btn-outline' %>
```
Comment (findandrew) [inline app/views/learn/_post_user_response.html.erb:8]: this may also have a url_helper

File: app/views/users/deck_preview.html.erb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                <%= pluralize(@deck.cards.count, 'Card') %>
+              </div>
+              <div>
+                *static* 9.5 Difficulty
+              </div>
+            </div>
+            <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
+          </div>
+          <div class="controls">
+            <%= link_to 'Begin -or- Resume', "/learn/#{@user.id}/#{@deck.id}/", remote: false, class: 'btn btn-brand btn-outline' %>
```
Comment (findandrew) [inline app/views/users/deck_preview.html.erb:32]: might have url_helper

File: app/views/users/deck_preview.html.erb [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<div class="page-learn">
+  <!-- Intro -->
+  <div id="deck-intro container-learn">
+    <div class="container">
+      <div class="row">
+        <div class="col-sm-12 col-md-8 col-md-offset-2 learn-panel intro">
+          <div class="header">
+            <div class="image">
+              <%=image_tag image_path("temp/dashboard-img-#{rand(1..4)}.jpg"), class: 'img-circle' %>
```
Comment (findandrew) [inline app/views/users/deck_preview.html.erb]: does this work without parens () ?

File: app/views/learn/_get_card_prompt.html.erb [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<div class="container">
+  <div class="row">
+    <div class="col-sm-12 col-md-8 col-md-offset-2 learn-panel prompt">
+      <div class="content">
+        <%= form_for(:card_response, remote: true, :url => "/learn/#{@user.id}/#{@deck.id}/post-user-response", :html => {:class => 'prompt-form'}) do |f| %>
```
Comment (findandrew) [inline app/views/learn/_get_card_prompt.html.erb]: does this work without parens () ?

File: app/views/users/_dashboard_deck.html.erb (hunk 2 above)
Comment (cstillitano) [inline app/views/users/_dashboard_deck.html.erb:8]: i don't see a url_helper for this or the rest of the helper comments below.
is it possible/best practice to create helpers for non-resourceful routes?
having some difficulty finding clear documentation on this ...

Comment (findandrew) [inline app/views/users/_dashboard_deck.html.erb:8]: ouuu thats right, its the resource that makes the helper not the routes itself. i confirmed my method doesnt have a helper either:

```
                         POST   /learn/:id/:deck_id(.:format)             learn#create_card_response
                   users GET    /users(.:format)                          users#index
```

Comment (findandrew) [inline app/views/users/_dashboard_deck.html.erb:8]: yeah dont worry about these comments :)

**Pull request merged**